opensrp-anc/src/main/java/org/smartregister/anc/library/presenter/ANCJsonFormFragmentPresenter.java

```diff
@@ -0,0 +1,136 @@
+import static org.smartregister.anc.library.constants.AncFormConstants.SpinnerKeyConstants.SUB_DISTRICT;
+import static org.smartregister.anc.library.constants.AncFormConstants.SpinnerKeyConstants.VILLAGE;
+import static org.smartregister.anc.library.widget.ANCSpinnerFactory.descendants;
+import static org.smartregister.anc.library.widget.ANCSpinnerFactory.locationSpinners;
+
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+
+import androidx.core.util.Pair;
+
+import com.google.gson.Gson;
+import com.vijay.jsonwizard.customviews.MaterialSpinner;
+import com.vijay.jsonwizard.interactors.JsonFormInteractor;
+import com.vijay.jsonwizard.presenters.JsonFormFragmentPresenter;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.smartregister.anc.library.R;
+import org.smartregister.anc.library.activity.AncRegistrationActivity;
+import org.smartregister.anc.library.fragment.ANCRegisterFormFragment;
+import org.smartregister.anc.library.util.Utils;
+import org.smartregister.domain.Location;
+import org.smartregister.util.JsonFormUtils;
+
+import java.util.List;
+
+import timber.log.Timber;
+
+public class ANCJsonFormFragmentPresenter extends JsonFormFragmentPresenter {
+
+    public static final String TAG = ANCJsonFormFragmentPresenter.class.getName();
+
+    private final AncRegistrationActivity jsonFormView;
+    private final ANCRegisterFormFragment formFragment;
+
+    public ANCJsonFormFragmentPresenter(ANCRegisterFormFragment formFragment, JsonFormInteractor jsonFormInteractor) {
+        super(formFragment, jsonFormInteractor);
+        this.formFragment = formFragment;
+        jsonFormView = (AncRegistrationActivity) formFragment.getActivity();
+    }
+
+    @Override
+    public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+        super.onItemSelected(parent, view, position, id);
+
+        String key = (String) parent.getTag(R.id.key);
+        String isHumanAction = String.valueOf(parent.getTag(R.id.is_human_action));
+
+        try {
+            if (locationSpinners.contains(key) && Boolean.parseBoolean(isHumanAction)) {
+
+                JSONObject field = JsonFormUtils.getFieldJSONObject(formFragment.getStep(STEP1).getJSONArray(FIELDS), key);
+                String parentLocationId = field.getString(VALUE);
+
+                if (key.equals(PROVINCE)) {
+                    populateLocationSpinner(parentLocationId, DISTRICT, descendants.get(DISTRICT));
+                } else if (key.equals(DISTRICT)) {
+                    populateLocationSpinner(parentLocationId, SUB_DISTRICT, descendants.get(SUB_DISTRICT));
+                } else if (key.equals(SUB_DISTRICT)) {
+                    populateLocationSpinner(parentLocationId, FACILITY, descendants.get(FACILITY));
+                } else if (key.equals(FACILITY) && position > -1) {
```

Comment:
Please use https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html for this `equals` check

Reply:
Done